extensions/json/src/main/java/io/deephaven/jsoningester/JSONToTableWriterAdapterBuilder.java

```diff
@@ -0,0 +1,503 @@
+package io.deephaven.jsoningester;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.deephaven.base.Pair;
+import io.deephaven.io.logger.Logger;
+import io.deephaven.tablelogger.TableWriter;
+import io.deephaven.util.annotations.ScriptApi;
+import org.apache.commons.lang3.mutable.MutableLong;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToIntFunction;
+import java.util.function.ToLongFunction;
+
+import static io.deephaven.jsoningester.JSONToTableWriterAdapter.SUBTABLE_RECORD_ID_COL;
+
+/**
+ * The builder configures a factory for StringToTableWriterAdapters that accept JSON strings and writes a table.
+ */
```

Comment:
Do the interfaces actually require creating strings? (I thought that the kafka one operated on bytes or input stream...)

I think forcing json to wash through strings (if that indeed is the case) is a design limitation. I'd love for this json parsing logic to be used by kafka json.